mailpoet/lib/Logging/LoggerFactory.php

```diff
@@ -82,6 +82,9 @@ public function getLogger($name = 'MailPoet', $attachProcessors = WP_DEBUG) {
         $this->loggerInstances[$name]->pushProcessor(new MemoryUsageProcessor());
       }
 
+      // Adds the plugin's versions to the log, we always want to see this
+      $this->loggerInstances[$name]->pushProcessor(new PluginVersionProcessor());
```

Comment:
One thing that it is not clear to me is why we don't want to always include the three processors that already existed and why we want to always have the `PluginVersionProcessor`? Asking just in case there is a scenario where the code expects to run without any processors. I'm not sure if that is the case or not.

Also, if now we always include `PluginVersionProcessor` maybe we should change the name of the variable `$attachProcessors` to `$attachOptionalProcessors` or something like that?

Reply:
Good question. I don't know. Originally, this was to save some data and only keep what is important in the log. 

For the `PluginVersion` I think it is useful to have it always not only when `WP_DEBUG` is enabled because we want to know when did the customer produce each log. But maybe it is not worth differentiating and we could include always everything. What do you think?

Reply:
I'm not sure. I'm inclined to think we should not change that unless we have a good reason to (for example, support tells us that always having this information available would be helpful) and we check that it won't add a significant overhead.

Reply:
This information is not for supporters. This is for L2. We already had this problem when a customer claimed there is an error but we fixed it already in the latest version. So we weren't sure if the problem was fixed or if the customer did not update when the error occurred for them. 

If we only include the plugin version when `WP_DEBUG` is true, we wouldn't know what happens.

Reply:
Sorry, my previous comment was not clear. I was not referring to putting the plugin processor behind the WP_DEBUG flag. I was referring to keeping the other processors behind the WP_DEBUG flag or not. That is the part that I'm not sure if we should change or not due to the reasons that I shared above.

Reply:
Sorry @rodrigoprimo I am quite lost after this long discussion. I am quite unsure what do you suggest. Is there anything I should change to get this Pull request moving, or do you think it is good as is?

Reply:
To try to summarize my suggestions:

- You asked me if I think we should always include all the processors regardless of the value of the `WP_DEBUG` and I'm inclined to think that we should not change this part of the logic and keep it as is.
- I'm suggesting we change the name of the variable `$attachProcessors` as now there is one processor that is always included. One possible replacement is `$attachOptionalProcessors`.